Bundler, Rspec upgrade, and some other changes

stve opened the pull request:
This is primarily some simple changes.  I've added bundler, removed jeweler, and upgraded dependencies to hashie 1.0.0 and rspec 2.5.0

The larger change was to adjust the locations of some files to be a bit more in line with how the Mash class is defined (basically hashie/rash instead of all changes encapsulated in rash.rb).  I've persisted rash.rb to require 'hashie/rash' for backwards compatability.

I know your readme says not to fudge with the gemspec, but I needed to in order to ditch jeweler.

Thanks.

stve committed:
added gemfile

stve committed:
removed jeweler and updated dependencies

stve committed:
use class defined version

stve committed:
don't require rubygems and require hashie/mash

stve committed:
properly namespace rash inside hashie

stve committed:
ensure backwards compatability

tcocca commented:
Did you actually change anything here functionality wise?  What is the reason to remove Jeweler?  I can understand the namespacing and that is a good change to be more in line with Hashie::Mash but I don't get the removal of Jeweler.  Please elaborate.

Glad somebody else found this useful (I hate camelCasing!)

stve commented:
I've used it plenty on a number of different projects.  Camel-case was driving me nuts too and I'm glad I stumbled across this  when I went looking for a solution some time back.

No, there were no functional changes, I did adjust the indentation in the the main class but didn't touch a line of actual code.  As for Jeweler, it isn't nearly as useful as it once was when github served gems and there was a need for it's utility.  Most of the value deriven from Jeweler has been replaced by Bundler's tasks now which I included in the Rakefile.

joshk commented on rash.gemspec:
```diff
@@ -1,58 +1,29 @@
 
 Gem::Specification.new do |s|
   s.name = %q{rash}
   s.version = "0.2.0"
```
This isn't needed anymore as Hashie::Rash::VERSION has been added a couple of lines down

joshk commented:
+1 to all the changes made in this pull request, especially the removal of "require 'rubygems'"!!

stve committed:
remove duplicate version assignment

stve committed:
use a relative path in the spec helper

stve committed:
fix the LoadError issue when doing bundle install

On a clean clone, bundle install fails with the following error:
There was a LoadError while evaluating rash.gemspec:
  no such file to load -- hashie/mash from
  /Users/steve/Projects/rash/rash.gemspec:3

instead of Hashie::Rash::VERSION, just use Rash::Version to get around
this dependency issue.

sferik commented:
:+1:

If these get merged and released, I can replace the custom Rash class in faraday_middleware, which gets used in many popular gems (e.g. the twitter gem).

tcocca commented:
Ok, I am planning on merging this stuff hopefully this weekend.  I guess ill
just release a 0.3 version after that. Thanks guys.

~ tom

On Apr 2, 2011 1:02 PM, "sferik" <
reply@reply.github.com>
wrote:

:+1:

If these get merged and released, I can replace the custom Rash class in
faraday_middleware, which gets used in many popular gems (e.g. the twitter
gem).

## 

Reply to this email directly or view it on GitHub:
https://github.com/tcocca/rash/pull/1#issuecomment-949208

tcocca commented:
Here ya go guys, get it while its hot.

gem install rash

Thanks for the contributions, I didn't even know anybody else was using this gem, glad you guys find it useful.

~ Tom

sferik commented:
Awesome! Thanks, Tom.

sferik commented:
By the way, have you talked to Michael about getting this behavior merged into a future version of Hashie?

tcocca commented:
A long time ago, see this issue:  https://github.com/intridea/hashie/issues/closed#issue/7

Michael and Mislav kinda shot it down, so I created a rash gem, it seemed silly at the time to have a full gem for such a simple thing but its been fine so far.  Maybe they'll link to it in the README for hashie.

What are you using rash on anyway?

sferik commented:
See the thread on this commit: https://github.com/pengwynn/faraday_middleware/commit/3130404a16022b36f3c9786fc05e6c75bfc0a746#comments

tcocca commented:
Cool, well let me know if there are any problems with the new stuff and I'll take a look.
